### ISSUE TITLE
lottie: override default slots

### ISSUE THREAD

**Issue by theashraf** [thorvg/thorvg#2953]
When applying the following slots JSON to the attached animation, the background color should update as specified, but it does not.

Using the `tvg_lottie_animation_override` API:
```json
{
  "bg_color": {
    "p": {
      "a": 0,
      "k": [
        1,
        0.8196,
        0.2275
      ]
    }
  },
  "check_color": {
    "p": {
      "a": 0,
      "k": [
        0.0078,
        0.0078,
        0.0078
      ]
    }
  }
}
```

**Actual Result:**
<img width="382" alt="image" src="https://github.com/user-attachments/assets/a508bb80-4dd7-4c1f-86e0-a37b466fc05a">

**Expected Result:**
<img width="384" alt="image" src="https://github.com/user-attachments/assets/2c2faaba-4d61-405d-99de-51f45285b754">

[animation_with_default_slots.json](https://github.com/user-attachments/files/17730431/animation_with_default_slots.json)

---

**Note:** The attached animation already has default slots:

```json
{
  "bg_color": {
    "p": {
      "a": 1,
      "k": [
        {
          "o": {
            "x": 0.167,
            "y": 0.167
          },
          "i": {
            "x": 0.833,
            "y": 0.833
          },
          "s": [
            0.2275,
            0.5255,
            1
          ],
          "t": 0
        },
        {
          "s": [
            0.7294,
            0.2275,
            1
          ],
          "t": 89
        }
      ]
    }
  },
  "check_color": {
    "p": {
      "a": 0,
      "k": [
        1,
        1,
        1
      ]
    }
  }
}
```

**Comment by theashraf** [comment]
Since this animation already has default slots, calling `tvg_lottie_animation_override` with a null pointer shouldn’t remove the default slots, right? It should only remove the user-provided overrides, not the default ones, correct ?

**Comment by tinyjin** [comment]
> Since this animation already has default slots, calling `tvg_lottie_animation_override` with a null pointer shouldn’t remove the default slots, right? It should only remove the user-provided overrides, not the default ones, correct ?

Yes default slots cannot touch `override` behavior. 

- Shouldn't be removed by `tvg_lottie_animation_override` with a null pointer 
- Shouldn't affect to following slots by user call

**Comment by tinyjin** [comment]
@theashraf Yes, after approval, we'll deliver in v0.15.x

**Comment by hermet** [comment]
@theashraf the fix will be in v0.15.4. Thanks.
